File: tests/fixtures/javascript/class_heavy.js
import { Base } from './base';

class Dog extends Base {
  constructor(name) {
    super();
    this.name = name;
  }

  sound() {
    return 'Woof';
  }

  async fetch() {
    return;
  }
}

class Cat {
  constructor(name) {
    this.name = name;
  }

  sound() {
    return 'Meow';
  }
}

File: tests/fixtures/python/class_heavy.py
"""Classes with methods, including inheritance and async methods."""
from abc import ABC, abstractmethod


class Animal(ABC):
    def __init__(self, name: str):
        self.name = name

    @abstractmethod
    def sound(self) -> str:
        pass


class Dog(Animal):
    def sound(self) -> str:
        return "Woof"

    def fetch(self) -> None:
        print(f"{self.name} fetches")


class Cat(Animal):
    def sound(self) -> str:
        return "Meow"

    async def nap(self, duration: int) -> None:
        pass
